### PR TITLE
feat: code style for cache package

### PR DESCRIPTION
**Pull request by daheige**
**Please provide issue(s) of this PR:**
Fixes https://github.com/polarismesh/polaris/issues/774
code style for cache package

### CODE REVIEW

**Comment by daheige** [comment]
Here is the save operation. Grouping is reasonable. After the previous function is executed immediately, there is a lock behind it. The save operation is likely to cause a longer lock waiting time.


------------------ Original ------------------
From: liaochuntao ***@***.***&gt;
Date: Thu,Oct 27,2022 11:28 PM
To: polarismesh/polaris ***@***.***&gt;
Cc: heige ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [polarismesh/polaris] feat: code style for cache package (PR#773)





 
@chuntaojun commented on this pull request.
 
 
In cache/config_file_bucket.go:
 &gt; @@ -66,19 +66,14 @@ func (b *configFileGroupBucket) saveGroupById(id uint64, item *model.ConfigFileG  }    func (b *configFileGroupBucket) saveGroup(namespace, group string, item *model.ConfigFileGroup) { -func() { -b.lock.Lock() -defer b.lock.Unlock() +b.lock.Lock() +defer b.lock.Unlock()  
锁的持有时间变长了

File: cache/config_file_bucket.go [hunk 1]
```diff
@@ -66,19 +66,14 @@ func (b *configFileGroupBucket) saveGroupById(id uint64, item *model.ConfigFileG
 }
 
 func (b *configFileGroupBucket) saveGroup(namespace, group string, item *model.ConfigFileGroup) {
-	func() {
-		b.lock.Lock()
-		defer b.lock.Unlock()
+	b.lock.Lock()
+	defer b.lock.Unlock()
```
[the remark this answers is not in the document]
Comment (chuntaojun) [reply cache/config_file_bucket.go:70]: > Here is the save operation. Grouping is reasonable. After the previous function is executed immediately, there is a lock behind it. The save operation is likely to cause a longer lock waiting time.
> […](#)


lock 处理的是 map[namespace]bucket 中关于bucket 的插入，一旦 bucket 插入完成，lock 的写锁本身就应该立即释放，而不是不断持有写锁直到 bucket 的插入操作完成

**Comment by chuntaojun** [comment]
> Here is the save operation. Grouping is reasonable. After the previous function is executed immediately, there is a lock behind it. The save operation is likely to cause a longer lock waiting time.
> […](#)

lock 处理的是 map[namespace]bucket 中关于bucket 的插入，一旦 bucket 插入完成，lock 的写锁本身就应该立即释放，而不是不断持有写锁直到 bucket 的插入操作完成

**Comment by daheige** [comment]
Before this, a function was executed immediately, and whether there was an initialization operation was judged in the immediately executed function. If there was no initialization, a lock operation was performed here. Then another lock operation is performed, the write operation is performed, and the lock is released. I adjusted the code so that it only needs to lock, unlock, initialize and write again once. For locks, it actually reduces the time and granularity of locking and unlocking. I will add a unit test case later, and you will understand why I do this.


------------------ Original ------------------
From: daheige ***@***.***&gt;
Date: Thu,Oct 27,2022 11:31 PM
To: reply+ACMGV64GPTR7VPAGFD2LHNOBM7LDPEVBNHHFJ6LAVQ ***@***.***&gt;, polarismesh/polaris ***@***.***&gt;
Cc: daheige ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [polarismesh/polaris] feat: code style for cache package (PR#773)



Here is the save operation. Grouping is reasonable. After the previous function is executed immediately, there is a lock behind it. The save operation is likely to cause a longer lock waiting time.


------------------ Original ------------------
From: liaochuntao ***@***.***&gt;
Date: Thu,Oct 27,2022 11:28 PM
To: polarismesh/polaris ***@***.***&gt;
Cc: heige ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [polarismesh/polaris] feat: code style for cache package (PR#773)





 
@chuntaojun commented on this pull request.
 
 
In cache/config_file_bucket.go:
 &gt; @@ -66,19 +66,14 @@ func (b *configFileGroupBucket) saveGroupById(id uint64, item *model.ConfigFileG  }    func (b *configFileGroupBucket) saveGroup(namespace, group string, item *model.ConfigFileGroup) { -func() { -b.lock.Lock() -defer b.lock.Unlock() +b.lock.Lock() +defer b.lock.Unlock()  
锁的持有时间变长了

**Review by chuntaojun — APPROVED** [review]

**Pull request merged**